### DISCUSSION TITLE
2024-10-9 編集会議

### DISCUSSION THREAD

**Discussion by kidayasuo** [w3c/jlreq-d#66] [General]
# 2024-10-9 編集会議
出席：敏先生、山口、下農、木田

- 「3.2 ベースラインシステムでの和文組版」をドラフトした（下）
- 1章および、2章2.2 の「ひらがなとカタカナ」までの文面をレビュー。次は2.2漢字から。
- 次回編集会議は10月22日火曜日9時から

---- 
## 3.2 ベースラインシステムと和文組版

3.1で述べたボディを配置する方法は、ベースラインを使う方法では次のように表すことができる。

ボディの高さの中心を通る線を基準にして、グリフを字幅に従って配列してゆく。また、目的により下端（縦組においては右端）または上端で揃える場合もり、その行の主となるフォントサイズにおけるボディの下端や上端を通るベースラインも必要となる。よって、和文組版においては中心、下端、上端、と合計三つのベースラインが存在することになる。

和文とアルファベティック・ベースラインを用いる欧文を混在させる方法は〓で説明する。

----

**Comment by KobayashiToshi** [comment]
第2章のタイトルの問題ですが，

それ以前の問題として，以下が第2章に含まれているが，
- 行そろえのスタイル
- ルビ
- 横書きと縦書き

これらは，第3章の“日本語組版の基本アーキテクチャ”に持っていった方がよくないか．その方がふさわしいのではないかな？　ただし，“日本語組版の基本アーキテクチャ”は，いくつか書き直しが必要になる．

なお，“ルビ”については内容が未定だが，ここでは，日本語組版には，ルビってものがあるよ，こんな使用例があるよ，その他，自動生成などいくつか．なお，配置の問題は，かなり複雑なので，後で説明するよ，ということでどうですか．“こんな使用例があるよ”は，“drafts”の“4_ルビの組版処理”の冒頭の部分にある“4.2　ルビの使用例”の内容でいいかとも思っている．

もし，上記を“日本語組版の基本アーキテクチャ”に持っていくのであれば，第2章のタイトルは，以下が考えられる
- 日本語組版で使用する文字
- 日本語組版で使用する文字の基礎知識

**Reply by kidayasuo** [reply]
2章は、ウェブページ向けにテキストを作ったり、ワープロなどを使う（そして必ずしも日本人でない）ユーザー向けで、その人たちが知っているべき基本的なことをカバーしています。ルビ、縦書き横書き、などの機能のあることやその用法はそこに必要だと思います。行揃えは、もし両端揃えと行頭揃えを段落の説明のところでカバーしてしまえば、残りは不要かもしれません。

レイアウトの詳細はブラウザやワープロの仕事ですので、3章以降で改めてレイアウトを説明することになります。

**Comment by KobayashiToshi** [comment]
和字のところだったと思いが，漢字と仮名で，単語が認識できる，という説明があったが，それだけで，十分だと思われるとよくないのではと，ちょっと気になる．

ただし，別語の漢字や仮名が連続して，あいまいになる例もあるが，こうした場合，読点などを補うことで，ほぼ問題なく認識できているように思う．この文の“…場合，読点…”が，まさにその例です．

そこで，“約物（読点といってもよい）などを併用することで……”みたいな文言を追加した方がよくないか．

なお，これは書かなくてもよいが，感想ですが，英語の場合，コンマの使用箇所は，日本語に比べ，かなりはっきりしていて，しかも，その有無が文の意味に影響することもあるように感じている．これに対して，日本語では，読点の使い方には，かなりのゆれというか，幅がある．ですので，この自由に使用できる読点が，漢字と仮名の使い分けだけではない，語句の区切りにけっこう役立っているようにも思います．

**Reply by kidayasuo** [reply]
[2.1 日本語書き言葉の特徴](https://github.com/w3c/jlreq-d/blob/gh-pages/drafts/2.%20日本語デジタルテキスト入門（Introduction%20to%20Japanese%20Digital%20Text）.md#21-日本語書き言葉の特徴) ですね。

概説は簡潔さが重要ですので、でどこまで精密に書く必要があるのかを考える必要があります。確かに、区切りを確実に示すわけではないので「明示的に語句を区切る文字がなくても語句のまとまりを認識できる」は、もう少し緩く「明示的に語句を区切る文字がなくてもおおよその語句のまとまりを認識できる」と「おおよそ」を付けるのはどうでしょう。

平安時代における仮名漢字混じり文の成立を考えると、読点は必須とは言えないでしょう。読みやすくするための工夫は、読点だけではなく、語句を漢字にしたり仮名に開いたりする、語彙自体を工夫する、助詞を省いたりあえて加えたり、などの手段があると思いますが、これらは触れるとしたら書き手のための2章がより適当？

**Comment by KobayashiToshi** [comment]
あまり細かく説明しないで，ラテン文字はベースライン，和字は違うよ，ということだけを書いたように記憶している．以下のように（ただし，内容は同じだげ文章表現はだいぶ違うかも）．なお，全幅の文字を密着させる場合だけでなく，アキ組でも，詰め組でもボディのY方向の中心をそろえる点は変わらない．また，“これら三つのベースラインと、欧文組版に用いるアルファベティック・ベースラインとの関係は後の説で説明する。”は必要か？　原案では書かなかったと思う．ここでは，違っているというだけでいいのではないか．

そこで，タイトルも“ベースラインシステムと和文組版”としたらどうか．

3.2 ベースラインシステムと和文組版

行に文字を配置していく場合，ラテン文字では，アルファベティック・ベースラインを基準に，その水平線に沿って文字を配置していく．これに対し，和文では，ボディのY方向の中心をそろえて配置するのが原則である．なお，異なる文字サイズの文字が混ざる場合もボディのY方向の中心をそろえる方法が一般的であるが，ボディの前端または後ろ端をそろえる方法にする場合もある．

もし，追記するなら，以下でよくないか．

なお，こうした異なる基準線の和字とラテン文字が混在する場合の配置方法の問題点は，〓〓で解説する．

> それらの機能が同一行に混在する場合、ここは真ん中、ここは下端、などとなるか？　それとも文章全体でスタイルとして統一か？

これは，文章全体でのスタイルとはならない．文字サイズを変える内容により，使い分ける．ですから指定対象は，段落でなく，文字列．〈p〉…〈/p〉でなく，〈span〉…〈/span〉．

>  OpenTypeの場合、ボディの高さはどうやって知る？

先日の会議では，私がしつこく山本さんに聞いたが，よくわからない．最初は方法があるように言っていたが，会議の終わりでは，特にラテン文字では，そもそもボディの高さなんて，その情報は必要ないといっていた．

和文とラテン文字について，再度，OpenTypeの場合，ボディの高さは，どうやって知ることができるか聞いてみる必要があるが，もし，ボディの高さが，ラテン文字も和文も，わからない場合は，原点をそろえるしかないが，よく考えてみると，和文は，字幅の情報をY方向も，X方向も持っているはずであるし，山本さんも，それはあるといっていた．これがないと縦組ではベタ組にならない．ということは，縦組では，横組用の字幅情報がボディの高さになり，横組では，縦組用の字幅情報がボディの高さになると思われるので，和文ではボディの高さを知ることができると思われる．

ですから，たぶん，和文ではボディの高さわかるが，ラテン文字ではボディの高さがわからないということは考えられる．

**Reply by kidayasuo** [reply]
三種類の基準線のあることは元々の文でも説明していました。ここの主眼は、ボディを配列してゆく方式から、基準線を使う方式への転換方法ですので、そのことの説明が第一です。欧文と異なる点に関しては、確かにこんなに固く書く必要はありませんね。またタイトルも了解です。書き直します。

**Reply by kidayasuo** [reply]
上の3.2をアップデートしました。いかがでしょう。

一つ疑問なのですが、文字の実際のインクとボディの枠の間に空間があるので、ボディの下端を基準とすると、大きなサイズの文字はその空間が大きいのでより上方に、小さなサイズの文字は空間が小さいのでより下方に位置することになります。つまり下端が揃いません。これはそれで良いのですか？　それとも、できればNatが言っていた平均字面の上端下端を使うべきでしょうか。もちろん活字では字面で揃えるなどという芸当は無理でしょうけれど、デジタルだと字面を合わせる方が適当な可能性がありますね。これは山本さんやNatさんとも相談でしょうか。

**Reply by kidayasuo** [reply]
三つのベースライン、という言葉は山本さんが説明されたもので、システムに対する要求事項がとてもわかりやすいという意味で優れた言い方だと思います。

**Comment by KobayashiToshi** [comment]
>2章は、ウェブページ向けにテキストを作ったり、ワープロなどを使う（そして必ずしも日本人でない）ユーザー向けで、その人たちが知っているべき基本的なことをカバーしています。ルビ、縦書き横書き、などの機能のあることやその用法はそこに必要だと思います。行揃えは、もし両端揃えと行頭揃えを段落の説明のところでカバーしてしまえば、残りは不要かもしれません。レイアウトの詳細はブラウザやワープロの仕事ですので、3章以降で改めてレイアウトを説明することになります。

そうだとは思いますが，レイアウトの詳細を解説する第3章は分離して，“第3章　日本語組版の基本アーキテクチャ”と独立の章になった．レイアウトの詳細は“第4章　行への文字の配置の実際”“第5章　ルビ（圏点、行間注）”“第6章　行の分割”…の構成に変わった．

ですので，第2章と第3章は，一般ユーザー向けの概説と考えていいようにも思うのです．現行の第3章は，多少は書き直す必要はありますが，レイアウトの詳細は書かれていませんし，日本語組版の最も必要となる事項がごく簡単に解説していますから，ここに第2章で文字でない事項を持ってくることは可能でしょうし，それは，“ウェブページ向けにテキストを作ったり、ワープロなどを使う（そして必ずしも日本人でない）ユーザー向け）に”なるように思うのです．

最も気になるのは，第2章に文字の部分と段落のスタイル等が一緒になっていることです．

ですので，第2章と第3章は，以下の3つが考えられる．
　第2章と第3章は，別々にする．
　第2章と第3章は合併して，中見出しで第2章と第3章は区別する．
　第2章と第3章は，別々にするが，例えば，見出しを以下のようにする．
　　第2章　日本語デジタルテキストの基礎1　文字とフォント
　　第3章　日本語デジタルテキストの基礎2　組版処理

22日（火）に議論しましょう．

**Reply by kidayasuo** [reply]
これはかなり大きな変更となりますので、ちょっと考えます。22日に議論しましょう。

**Reply by kidayasuo** [reply]
敏先生の提案は良いアイディアのように思えてきました。つまり、現在の2章は文字にフォーカスして、3章でユーザーが知っているべき組版の基礎を説明する。4章以降はエンジニア向け。今日あとで議論しましょう。

**Comment by KobayashiToshi** [comment]
>一つ疑問なのですが、文字の実際のインクとボディの枠の間に空間があるので、ボディの下端を基準とすると、大きなサイズの文字はその空間が大きいのでより上方に、小さなサイズの文字は空間が小さいのでより下方に位置することになります。つまり下端が揃いません。これはそれで良いのですか？　それとも、できればNatが言っていた平均字面の上端下端を使うべきでしょうか。

こうした問題は，文字サイズが異なる場合だけでなく，同一のサイズの場合も含め，いつでも発生する問題．どこまで対応するかは，一概にいえない．

考え方としては，どれだけ問題が深刻か，それへの対処に手間暇をかけられるかという判断の問題に関係する．書籍を例にとると，
　表紙など　徹底的に手間暇かけて，極端の場合は，上下・左右の位置やアキだけでなく，字面もアウトライン化して，修正するかもしれない（過去にやった場合もあった）．
　見出しなど　ある程度の機械的な処理が可能な場合は，本文とは異なる処理を選択する場合もある．私のよくやる例は，括弧や句読点のアキが二分ではアキ過ぎなので，四分アキに詰める方法は，よくやっている．
　本文は，パーレン等で，ベタ組にするという例外的に処理する方法もあるが，原則としては，決めた方針で機械的に処理している．

そもそも，行中で文字サイズを大きくする例は，最近，たまに見掛けるが，多くは，強調で，1字ではなく，文字列のことが多く，しかも，センターそろえが多い．上端，下端をそろえる場合は，文字サイズを小さくする場合なので，それほど気にならないし，平均字面の上端下端にしても，それほど大きな違いはないのではないかな．平仮名の端をそろえるというのは，ケースごとに考えないと，解決はむつかしいことにようにも思います．．

**Reply by kidayasuo** [reply]
確かに行頭の問題に比べてそれほど目立たないのかもしれませんね。了解です。

**Reply by kidayasuo** [reply]
もし敏先生の違和感が、2章に段落など組版が入っていることなら、下のような構成はどうでしょう？

3章は**仕組み**を述べているのでユーザー向けではないように思います。例えば「ベタ」はブラウザやワープロが何も言わなくてもやってくれます。ただ、空ける、詰める、などはワープロやウェブでユーザーが操作可能ですので、これらをユーザー向けとして説明しても良いかもしれませんね。

### 2　日本語デジタルテキスト入門　文字とフォント
現在の2章の文字とフォントの部分

### 3　日本語デジタルテキスト入門　文を作る
下記の内容を2章から分離する
- 段落と行そろえのスタイル
- ルビ
- 横書きと縦書き

### 4　日本語デジタルテキストの組版
現在の3章と4章

**Comment by KobayashiToshi** [comment]
>上の3.2をアップデートしました。いかがでしょう。

いいと思います．ここは，木田さんに納得してもらうことがまず第一ですから．

なお，

> 下端（縦組においては右端）または上端で揃える場合

下端または上端（縦組においては右端）で揃える場合

でないかな．

**Comment by KobayashiToshi** [comment]
>2.1 日本語書き言葉の特徴 ですね。
>「明示的に語句を区切る文字がなくても語句のまとまりを認識できる」は、もう少し緩く「明示的に語句を区切る文字がなくてもおおよその語句のまとまりを認識できる」と「おおよそ」を付けるのはどうでしょう。

語句の区別が，かなと漢字で完璧にできると思われては困るという点が気になっただけですので，上記でいいと思います．

**Comment by KobayashiToshi** [comment]
>これはかなり大きな変更となりますので、ちょっと考えます。22日に議論しましょう。

そうしましょう．ただ，1点追加すると，“第3章　日本語組版の基本アーキテクチャ”に書かれている事項は，“ウェブページ向けにテキストを作ったり、ワープロなどを使う（そして必ずしも日本人でない）ユーザー向け）に”とっても，字間は原則的にベタでいいよ，2行にわたる分割は，問題箇所はあるが，通常な漢字や仮名の部分で分割できるよ，特に行間の問題など，重要な情報のように思います．

**Reply by kidayasuo** [reply]
字間ベタや分割はワープロが自動的にやってくれるので、ユーザーが気にする必要はあるでしょうか？　行間はワープロのデフォルトがよくないことも多いので、書いておくと良いと思いますが。

**Comment by KobayashiToshi** [comment]
>これはかなり大きな変更となりますので、ちょっと考えます。22日に議論しましょう。

この点について，もう1点補足，

第2章の後半と第3章の下は，内容は重なっています．そして，どちらを主に解説するのか，あるいは，その2つを別々に説明するのか（これはたぶん無理だと思う），それとも同時に説明し，主にaというか，それとも主にbというか．ここではbだけどaでもあるよというか，aだけどbでもあるよというか，中立的にいい，ニュアンスは見出しで表現するか，考えないといけないことは多い．

a “ウェブページ向けにテキストを作ったり、ワープロなどを使う（そして必ずしも日本人でない）ユーザー向け）に”

b 日本語組版の基本アーキテクチャ
